.github/workflows/cmake-multi-platform.yml

```diff
@@ -0,0 +1,172 @@
+      with:
+        build-dir: ${{ github.workspace }}/build
+        source-dir: ${{ github.workspace }}
+        cc: gcc
+        cxx: g++
+        build-type: ${{ matrix.buildType }}
+        run-test: false # true
+        ctest-options: ${{ env.CTEST_OPTIONS }}
+
+  fedora-gcc-sanitizer:
+    strategy:
+      matrix:
+        buildType: [UdefSan, LeakSan, AddrSan, ThreadSan]
+    runs-on: ubuntu-latest
+    container: "fedora:latest"
+    steps:
+    - name: Update package list
+      run: dnf update -y
+    - name: Install Dependencies
+      run: |
+        sudo dnf install -y ${{ env.BUILD_DEPENDENCIES }}
+        sudo dnf install -y ${{ env.OSTREE_TUI_DEPENDENCIES }}
+        sudo dnf install -y libasan libubsan liblsan libtsan
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: "Build" # & Test
+      env:
+        CPR_BUILD_TESTS: OFF
+      uses: ashutoshvarma/action-cmake-build@master
+      with:
+        build-dir: ${{ github.workspace }}/build
+        source-dir: ${{ github.workspace }}
+        cc: gcc
+        cxx: g++
+        build-type: ${{ matrix.buildType }}
+        run-test: false # true
+        ctest-options: ${{ env.CTEST_OPTIONS }}
+
+  clang-tidy:
+    strategy:
+      matrix:
+        container: ["fedora:latest"]
+        buildType: [Debug]
+    runs-on: ubuntu-latest
+    container: ${{ matrix.container }}
+    steps:
+    - name: Update package list
+      run: sudo dnf update -y
+    - name: Install dependencies
+      run: |
+        sudo dnf install -y cmake git gcc-c++ clang ninja-build automake autoconf clang-tools-extra
+        sudo dnf install -y ${{ env.OSTREE_TUI_DEPENDENCIES }}
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: "Build Debug" # & Test"
+      env:
+        CPR_BUILD_TESTS: OFF # ON
+      uses: ashutoshvarma/action-cmake-build@master
+      with:
+        build-dir: ${{github.workspace}}/build
+        source-dir: ${{github.workspace}}
+        cc: clang
+        cxx: clang++
+        build-type: ${{ matrix.buildType }}
+        run-test: false
+        configure-options: -DCPR_ENABLE_LINTING=ON
+
+  clang-format:
+    runs-on: ubuntu-latest
+    container: fedora:latest
+    steps:
+    - name: Update package list
+      run: sudo dnf update -y
+    - name: Install clang-format
+      run: sudo dnf install -y clang-tools-extra
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Check format
+      run: bash scripts/check_clang_format.sh
+
+  cppcheck:
+    strategy:
+      matrix:
+        container: ["fedora:latest"]
+        buildType: [Release]
+    runs-on: ubuntu-latest
+    container: ${{ matrix.container }}
+    steps:
+    - name: Update package list
+      run: dnf update -y
+    - name: Install Dependencies
+      run: |
+        sudo dnf install -y ${{ env.BUILD_DEPENDENCIES }}
+        sudo dnf install -y ${{ env.OSTREE_TUI_DEPENDENCIES }}
+        sudo dnf install -y cppcheck
```

Comment:
cppcheck is installed but not used? Where do you set cppcheck enabled in CMake?

Reply:
Thanks for pointing that out! I now fixed it by just letting it run in a pre-defined Github action, making a few things a lot easier.